lib/ClangImporter/ClangModuleDependencyScanner.cpp

```diff
@@ -175,18 +181,89 @@ void ClangImporter::recordModuleDependencies(
     // Swift frontend option for input file path (Foo.modulemap).
     swiftArgs.push_back(clangModuleDep.ClangModuleMapFile);
 
+    // Handle VFSOverlay.
+    if (!ctx.SearchPathOpts.VFSOverlayFiles.empty()) {
+      for (auto &overlay : ctx.SearchPathOpts.VFSOverlayFiles) {
+        swiftArgs.push_back("-vfsoverlay");
+        swiftArgs.push_back(overlay);
+      }
+    } else {
+      // HACK: find the -ivfsoverlay option from clang scanner and pass to
```

Comment:
Could you please explain the reasoning in a comment? How/Why could there be overlays here not covered by the above `if` conditional?

Reply:
I forgot about this hack completely. This is rdar://105684525